### ISSUE TITLE
Released/ Unreleased. 

### ISSUE THREAD

**Issue by adamosborne83** [jurialmunkey/skin.arctic.fuse#999]
### Skin section

Widgets

### Current Behavior

Hi J, I don't know if this is a bug or a request.
On tv shows, land scape, if the show is not released it sometimes will either show episode artwork, or a fall back of whatever the image might be.

This is not helpful with watch next or when I go into a season, as it could look like I have 6 episodes to watch but it turns out to be only 1.

I know I can scroll to see the release date, but I am on about visually first looks.

I know on addons you can have options for unreleased episodes to be red, but this this not transpire to skin.

### Expected Behavior

Visually see released/ unreleased episodes 

### Steps To Reproduce

Go to series and open

### Screenshots and Additional Info

![Screenshot_20240712_154100_Discord](https://github.com/user-attachments/assets/15c07104-7061-4737-b764-6545e1eda0f6)
Episode 9 has not been released, but with the episode art if seems like it has.

Either a red box around it? Or some sort of identifier to show its unreleased? 

### Checklist

- [X] I have made a separate issue for each bug that I am reporting.
- [X] I have searched the issues and this bug has not been reported.
- [X] I am using the most recent skin version.
- [X] I have confirmed that this bug does not occur in the default skin.

**Comment by L0bsang** [comment]
I mean, the episode release date is right there. 07/18/2024. 

If that date is in the future, then the episode is unreleased right? Thumbnails and certain information seem to be available before the episode releases, I see that all the time on TMDB and IMDB.

Or am I missing something about your request? Date not sufficient and you want the colour changed as well?

>I know on addons you can have options for unreleased episodes to be red, but this this not transpire to skin.
I thought it did. Maybe it doesn't apply to homescreen widgets.

**Comment by adamosborne83** [comment]
> I mean, the episode release date is right there in your image: 07/18/2024. 
> absolutely, as I mentioned I know I can scroll across to see the release date, but this is all about visual and the skin not replicating what would be shown on estuary.
> If that date is in the future, then the episode is unreleased right? Thumbnails and certain information are usually available before the episode releases, I see that all the time on TMDB and IMDB for airing shows.
> information is fine for synopsis,  but I am talking about the landscape image. 
This is all about the image or look, so either a RED box around it (or whatever ever colour you select in addon)
> Or am I missing something about your request? Date not sufficient and you want the colour changed as well?
> scenario is, up next from trakt, 8 episodes left, so all show art work of some sort, but it looks like you have multiple episodes to watch, when in fact you have 1. 
> >I know on addons you can have options for unreleased episodes to be red, but this this not transpire to skin.
> 
> I thought it did (edit: definitely does within the addon I use). Although maybe it doesn't apply to homescreen widgets. 
> well image I attached shows this is not the case.
> However your image is not from the homescreen, so I would encourage you to check the Umbrella settings to see if you can colorize unreleased episodes to your liking.
This is red in my set up

**Comment by L0bsang** [comment]
> well image I attached shows this is not the case.
> This is red in my set up

Works fine on mine.

![image](https://github.com/user-attachments/assets/c20e95ca-df2f-4906-a954-34c92f0299b6)

**Comment by xyzfre** [comment]
![image](https://github.com/user-attachments/assets/c803de1b-45e5-48e1-b9f7-d41ea88b2e19) Also working on my end.

**Comment by jurialmunkey** [comment]
This is not something I can control directly in the skin. The addon/library is what supplies the links to the artwork (and in the screenshots above is also what is making the titles red by adding color tags to the label).

As to the skin adding some sort of indicator itself, it can't because there's no string or boolean condition to determine if an episode has aired and the skin engine itself can't do date comparisons.

**Comment by adamosborne83** [comment]
> > well image I attached shows this is not the case.
> >
> > This is red in my set up
> 
> Works fine on mine.
> 
> ![image](https://github.com/user-attachments/assets/c20e95ca-df2f-4906-a954-34c92f0299b6)
> 

Hmmmm wonder why mine doesn't

**Comment by OfficerKD637** [comment]
> This is not something I can control directly in the skin. The addon/library is what supplies the links to the artwork (and in the screenshots above is also what is making the titles red by adding color tags to the label).
> 
> 
> 
> As to the skin adding some sort of indicator itself, it can't because there's no string or boolean condition to determine if an episode has aired and the skin engine itself can't do date comparisons.

That makes sense. Do you think this feature could be added to TMDb Helper? Since it's so well integrated with AF, it'd maintain uniformity and make everything cohesive when using TMDbH widgets.

**Comment by adamosborne83** [comment]
> > well image I attached shows this is not the case.
> >
> > This is red in my set up
> 
> Works fine on mine.
> 
> ![image](https://github.com/user-attachments/assets/c20e95ca-df2f-4906-a954-34c92f0299b6)
> 

What addon you using?

**Comment by Nuklear92** [comment]
> > This is not something I can control directly in the skin. The addon/library is what supplies the links to the artwork (and in the screenshots above is also what is making the titles red by adding color tags to the label).
> > 
> > 
> > 
> > As to the skin adding some sort of indicator itself, it can't because there's no string or boolean condition to determine if an episode has aired and the skin engine itself can't do date comparisons.
> 
> That makes sense. Do you think this feature could be added to TMDb Helper? Since it's so well integrated with AF, it'd maintain uniformity and make everything cohesive when using TMDbH widgets.

I second this, I was also wondering the other day why the **Season Premiere/Finale** isn't working with TMDb Helper but works with other add-ons. Well, NOT really all add-ons as it also needs to be implemented by the developer of the add-on. Got me thinking perhaps @jurialmunkey did got the code into the Arctic Fuse skin, but not in the TMDb Helper add-on in particular. 🤔

**Comment by jurialmunkey** [comment]
> 
> I second this, I was also wondering the other day why the **Season Premiere/Finale** isn't working with TMDb Helper

It does work, but only for Trakt sync lists (e.g. Next Episodes or Calendar) as that's where the data comes from and I don't think it's worthwhile doing a separate call to check in other places -- I think it's a bit pointless in a normal season folder because it should be pretty clear which episode is first and which is last (hint: they're the ones at either end of the list!).

**Comment by OfficerKD637** [comment]
Sorry I think my first comment made it a bit confusing but I was asking if we could have something in TMDbH similar to those add-ons (colored + italicized labels) to identify unreleased episodes. 
Doesn't have to be red if you think it makes it look ugly. Can be an entirely different approach but some kind of a visual differentiator would be cool.

Let me know what you think!

**Comment by jurialmunkey** [comment]
> Sorry I think my first comment made it a bit confusing but I was asking if we could have something in TMDbH similar to those add-ons (colored + italicized labels, not the Season Premiere/Finale tags as those already work as expected) to identify unreleased episodes. 

I was responding to a different comment.

But... TMDbHelper does already do this too, but only for ListItem.Label whilst Landscape uses ListItem.Title -- and the whole noise about episode numbers in the landscape labels is exactly why it uses title not label. I can use ListItem.Label for Landscape but then people will complain that episode numbers are displayed there for some content.

**Comment by OfficerKD637** [comment]
> I was responding to a different comment.

Oh sorry about that! I got a notification and forgot to check the main comment.


> But... TMDbHelper does already do this too, but only for ListItem.Label whilst Landscape uses ListItem.Title -- and the whole noise about episode numbers in the landscape labels is exactly why it uses title not label. I can use ListItem.Label for Landscape but then people will complain that episode numbers are displayed there for some content.
> 
> 

Hmmm I apologize in advance if this is a dumb question, but is it possible to edit or add a condition so that ListItem.Title has colored/italicized text (similar to ListItem.Label) but only when the item is an episode (ListItem.DBType,episode maybe?)

**Comment by jurialmunkey** [comment]
> Hmmm I apologize in advance if this is a dumb question, but is it possible to edit or add a condition so that ListItem.Title has colored/italicized text (similar to ListItem.Label) but only when the item is an episode (ListItem.DBType,episode maybe?)

@OfficerKD637 - The reason ListItem.Title is not formatted by TMDbHelper is so that it is possible for skins to be able to have unformatted text. 

Formatting the label is a hack which will break if the skin is using its own inline formatting tags. That's why IMHO it should only be limited to ListItem.Label where it is far less likely a skin will try to use its own tags

For example, if the skin tries to apply italics like so `<label>[I]$INFO[ListItem.Title][/I]</label>` but the plugin forced its own tags into the title as a hack, then these will break because it will effectively be evaluated as `[I][COLOR=red][I]$INFO[ListItem.Title][/I][/COLOR][/I]` which is invalid (because the first closing tag closes both `[I]` tags) -- the result is a very ugly tag left in the label:
![image](https://github.com/user-attachments/assets/572ba42d-2a83-4d7f-9614-2c1f573777a6)

**Comment by OfficerKD637** [comment]
Damn it... I completely forgot there are so many other skins out there that use TMDb Helper and not just Fuse (major tunnel vision haha).
Even if there's a way (would probably be too convoluted) to add this as an experimental feature inside TMDb Helper only for Fuse users (maybe the addon reads the guisettings.xml or xbmc.getSkinDir() ?), it probably wouldn't make sense (to quote a wise man👀, it's like buying something you can't afford on credit... you pay for it later)

I'm not even sure if an addon can have exclusive/experimental features depending on which skin is active, but that's a whole new can of worms. But since both the skin and the addon are developed by you, it could mean adding new features without affecting other skins?

Feel free to tell me to shut up if all of this sounds silly.

**Comment by L0bsang** [comment]
> > > well image I attached shows this is not the case.
> > > This is red in my set up
> > 
> > 
> > Works fine on mine.
> > ![image](https://private-user-images.githubusercontent.com/162974281/348420931-c20e95ca-df2f-4906-a954-34c92f0299b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNjA5OTQsIm5iZiI6MTcyMTA2MDY5NCwicGF0aCI6Ii8xNjI5NzQyODEvMzQ4NDIwOTMxLWMyMGU5NWNhLWRmMmYtNDkwNi1hOTU0LTM0YzkyZjAyOTliNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNVQxNjI0NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZWI0OWQ1ZGExNGUxNTdkZWUwNmFjNWFhY2MyMmE4MjViNjQ3N2E5Y2FlZTg2NGMwNGUzN2FjNzRiMDBmOGM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.B5IXsc3Wl-wbtfEBag0kCA2cNxJpOZtnyy562fQBMlw)
> 
> What addon you using?

Just saw this, was busy on the weekend. I am using Fen.

**Comment by adamosborne83** [comment]
> > > > well image I attached shows this is not the case.
> > > > This is red in my set up
> > > 
> > > 
> > > Works fine on mine.
> > > ![image](https://private-user-images.githubusercontent.com/162974281/348420931-c20e95ca-df2f-4906-a954-34c92f0299b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNjA5OTQsIm5iZiI6MTcyMTA2MDY5NCwicGF0aCI6Ii8xNjI5NzQyODEvMzQ4NDIwOTMxLWMyMGU5NWNhLWRmMmYtNDkwNi1hOTU0LTM0YzkyZjAyOTliNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNVQxNjI0NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZWI0OWQ1ZGExNGUxNTdkZWUwNmFjNWFhY2MyMmE4MjViNjQ3N2E5Y2FlZTg2NGMwNGUzN2FjNzRiMDBmOGM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.B5IXsc3Wl-wbtfEBag0kCA2cNxJpOZtnyy562fQBMlw)
> > 
> > What addon you using?
> 
> @adamosborne83, Just saw this, was busy on the weekend. I am using Fen.

Cheers for this. I use umbrella @jurialmunkey would there be a reason that dependant on view type dictates if it shows red or not?

![20240715_165949.jpg](https://github.com/user-attachments/assets/d6c78c63-724a-4cf9-85ea-7f8bef8a4456)

**Comment by OfficerKD637** [comment]
> Cheers for this. I use umbrella @jurialmunkey would there be a reason that dependant on view type dictates if it shows red or not?
> 
> ![20240715_165949.jpg](https://github.com/user-attachments/assets/d6c78c63-724a-4cf9-85ea-7f8bef8a4456)
> 
> 

Yep, he mentioned that Landscape view uses ListItem.Title whereas the other view types use ListItem.Label. 
You can read his comment above on why TMDb Helper only formats labels and not titles (tl;dr TMDb Helper leaves the title unformatted so it doesn't break other skins. Labels on the other hand have a significantly lower chance of having custom tags)

**Comment by jurialmunkey** [comment]
> > > 
> > 
> > 
> > > I second this, I was also wondering the other day why the **Season Premiere/Finale** isn't working with TMDb Helper
> > 
> > 
> > It does work, but only for Trakt sync lists (e.g. Next Episodes or Calendar) as that's where the data comes from and I don't think it's worthwhile doing a separate call to check in other places -- I think it's a bit pointless in a normal season folder because it should be pretty clear which episode is first and which is last (hint: they're the ones at either end of the list!).
> 
> No, the Next Episodes doesn't show the Season Premiere/Finale labels. Which is the reason I made the comment in the first place. 😅 I do have the option turned on within the Skin settings, also I'm signed in with my Trakt account within TMDb Helper as well. I always find it odd that it never displays these labels within that list. 🤔

@Nuklear92 it definitely does
The feature did not exist in the skin until it was added in tmdbhelper and it was only after it was added that other addons started using it. It is not a standard kodi property - I literally invented the name of the property to use in tmdbhelper so the flags/banners could be displayed in the skin for tmdbhelper widgets. 

I use the next episodes widget all the time (honestly I probably watch at least 90% of all episodes via it) so I guarantee it definitely works (I'm literally looking at it on my TV now).

Check that you have enabled the option in tmdbhelper settings to allow it to use trakt calendar for next episodes (and also make sure you turned them on in the indicator skin settings).

**Comment by Nuklear92** [comment]
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > I second this, I was also wondering the other day why the **Season Premiere/Finale** isn't working with TMDb Helper
> 
> > > 
> 
> > > 
> 
> > > It does work, but only for Trakt sync lists (e.g. Next Episodes or Calendar) as that's where the data comes from and I don't think it's worthwhile doing a separate call to check in other places -- I think it's a bit pointless in a normal season folder because it should be pretty clear which episode is first and which is last (hint: they're the ones at either end of the list!).
> 
> > 
> 
> > No, the Next Episodes doesn't show the Season Premiere/Finale labels. Which is the reason I made the comment in the first place. 😅 I do have the option turned on within the Skin settings, also I'm signed in with my Trakt account within TMDb Helper as well. I always find it odd that it never displays these labels within that list. 🤔
> 
> 
> 
> @Nuklear92 it definitely does
> 
> The feature did not exist in the skin until it was added in tmdbhelper and it was only after it was added that other addons started using it. It is not a standard kodi property - I literally invented the name of the property to use in tmdbhelper so the flags/banners could be displayed in the skin for tmdbhelper widgets. 
> 
> 
> 
> I use the next episodes widget all the time (honestly I probably watch at least 90% of all episodes via it) so I guarantee it definitely works (I'm literally looking at it on my TV now).
> 
> 
> 
> Check that you have enabled the option in tmdbhelper settings to allow it to use trakt calendar for next episodes

Well, I'll double check again and see if the option from the tmdb helper settings are correct, I know from the skin side is turned on. As the feature does work with other add-ons I have installed. 

@jurialmunkey have a little request since related to tmdb helper which I honestly didn't wanted to open a whole new issue just for it. And it was to ask you, if it's possible to have an option to turn off that "dummy Tmdb video" every time you want to play a file using players. I know you can lower the time to the lowest "0.1 seconds" I believe, however the option is still on. I was wondering if you could make that dummy video be completely off optionally. Or is there a downside by doing this, either way I would take it as long as you make the option available to go further down the 0.1 seconds to fully OFF. 🤔

**Comment by jurialmunkey** [comment]
@Nuklear92 there is an option in expert settings. The downside is that any player which properly uses setresolvedurl will no longer pass resume points because that's how kodi manages files. The other option is to use players for addons which properly resolve via Kodi's setresolvedurl method and then there is no dummy file because it resolves to the correct file instead.

**Comment by Nuklear92** [comment]
@jurialmunkey Doubled checked my settings for Skin, Trakt and Widgets. Here are the screenshots below, the last one is inside the TMDB Helper, Your Next Episodes from Trakt menu. As you'll see all the episodes visible there are season premieres which in other addons that I have installed shows the **Season Premiere** label. Just not in the TMDB helper addon. I also tried deleting Trakt, TMDB, item details cache and nothing seems to make labels show. Any suggestions will be appreciated. 

![image](https://github.com/user-attachments/assets/aee6ff13-4e8b-46c8-964b-3be88b3b218f)
![image](https://github.com/user-attachments/assets/95ae47bf-728b-4455-aad7-59b492cb4dc0)
![image](https://github.com/user-attachments/assets/93272c1c-6649-4eb9-8498-e2041a27a991)
![image](https://github.com/user-attachments/assets/393df5b5-0fcc-4480-8a6f-f8700c7966a1)
![image](https://github.com/user-attachments/assets/2dbfe374-f127-43d9-af70-8b5a2dd9085d)

**Comment by OfficerKD637** [comment]
@Nuklear92 I think the setting you're looking for is in Details > Customize Indicators > Premiere / Finale (Calendar)
![image](https://github.com/user-attachments/assets/df036c0d-28a7-4faa-9087-b7b6864f536e)

**Comment by Nuklear92** [comment]
> @Nuklear92 I think the setting you're looking for is in Details > Customize Indicators > Premiere / Finale (Calendar)
> 
> ![image](https://github.com/user-attachments/assets/df036c0d-28a7-4faa-9087-b7b6864f536e)
> 
> 

![image](https://github.com/user-attachments/assets/5a699616-aeea-469a-b352-0f88c706dd35)

I have that setting on since the day he released that feature, as I was the one who created that issue request originally.

**Comment by adamosborne83** [comment]
Just an FYI, Umbrella dev has now made it that is an episode has not been released yet, it will just show the Fanart, not episode artwork. So this helps if Red writing is not showing, gets around skin doing listitem.title

Every little helps.

![526c44bb-c284-4fad-b3ec-bc274510d10f.jpg](https://github.com/user-attachments/assets/cf475a6d-afab-4a35-9030-2b65cf012bdc)

**Comment by jurialmunkey** [comment]
@Nuklear92 change sort in tmdbhelper settings to anything other than recently watched. Recent only gets the basic bare minimum sync data whereas the others get the extended data - I'm pretty sure that's the issue

**Comment by Nuklear92** [comment]
> @Nuklear92 change sort in tmdbhelper settings to anything other than recently watched. Recent only gets the basic bare minimum sync data whereas the others get the extended data - I'm pretty sure that's the issue

But just to make sure, are you getting the season premiere & season finale banners with the landscape thumbnails or only in text format near the plot?

**Comment by jurialmunkey** [comment]
> But just to make sure, are you getting the season premiere & season finale banners with the landscape thumbnails or only in text format near the plot?

@Nuklear92 - Both (assuming both settings are enabled for indicators and plot - note these are separate settings in skin).

![screenshot00609](https://github.com/user-attachments/assets/be571e41-bec4-44b0-b42f-7a5a38c06eec)

**Comment by jurialmunkey** [comment]
> Do you mind showing me what are your settings inside TMDB helper for Trakt, General, Widgets, etc. just to see if I have something turned ON that I'm not supposed in order to match yours and see if I can get it working.

@Nuklear92 - Sure. It's nothing special.

The only settings which should matter are
```
Widgets > Next episodes sort method > todays or lastweek (fairly sure airdate works as well but didn't check)
Widgets > Next episodes checks calender > Enabled
```

![screenshot00615](https://github.com/user-attachments/assets/98e25fc5-caac-4318-9aa2-d88991100288)

Possibly also need to enable this setting but it really shouldn't matter because its forced to be used regardless for Trakt content anyway.
```
General > Get additional details and artwork from TMDb > Enabled
```

![screenshot00617](https://github.com/user-attachments/assets/28fd670f-782d-4401-985a-49066d51f87e)

And here's my Trakt settings - nothing special.
![screenshot00614](https://github.com/user-attachments/assets/73b0fa83-8ac9-467f-bed9-8562b8d23d51)

**Comment by Nuklear92** [comment]
@jurialmunkey holy crap! You won't believe which setting was the one causing the labels **not** showing... 

As you described; "Next episodes sort method"... whenever you set that to anything but **recentlywatched** the labels will show. I find this odd that they will **not** show if you set the sort method as recently watched which is the default way from Trakt "Next-Up" within their website. 

Either way, man! I feel so stupid now, that I even ended up completely reinstalling Kodi from scratch and setting everything up to the way I had it just to now find out this simple setting was the one causing all that headache... 😵‍💫😂

Is this something that can be fixed/improved or that's just the way TMDB helper works with Trakt in the **your next episodes** list. That it won't display "everything" in this case the **season premiere/finale** labels when the user sets the sort as **recentlywatched**?
